src/pages/VenuePage/VenuePage.tsx

```diff
@@ -180,8 +187,15 @@ const VenuePage: React.FC = () => {
     return <LoadingPage />;
   }
 
-  if (!user) {
-    return <Login venue={venue} />;
+  if (!user || isLoginLoadingShown) {
+    return (
+      <Login
+        venue={venue}
+        isLoadingShown={isLoginLoadingShown}
```

Comment:
Do we need to pass these controls through like this? What happens if we only show the loading state within that component itself? Why does the loading page need to be controlled out here at this component’s level?

Reply:
That's because of the initial idea. My first approach was to just create a user + update his profile based on the data provided, but it was a unachievable because of the current implementation (user auth is checked higher in the hierarchy which causes `VenuePage` removal from the tree and addition afterwards)

Reply:
Currently, I don't think we need it to be on the root level though, will move it to the local one

Reply:
Removed in https://github.com/sparkletown/sparkle/pull/1667/commits/15dfaa4a6348a87a57db3386e982e12358de521c